src/pet.js

```diff
@@ -0,0 +1,76 @@
+const minFitness = 0;
+const minAcceptableFitness = 3;
+const maxFitness = 10;
+const minHunger = 0;
+const minAcceptableHunger = 5;
+const maxHunger = 10;
+const maxAge = 30;
+
+function Pet(name) {
+  this.name = name;
+  this.age = 0;
+  this.hunger = minHunger;
+  this.fitness = maxFitness;
+}
+
+Pet.prototype = {
+  get isAlive() {
+    return (
+      this.age < maxAge &&
+      this.hunger < maxHunger &&
+      this.fitness > minFitness
+    );
+  },
+};
+
+Pet.prototype.growUp = function () {
+  if (!this.isAlive) {
+    throw new Error("(*_*) Your pet has died.");
+  }
+  this.age += 1;
+  this.hunger += 5;
+  this.fitness -= 3;
+};
+
+Pet.prototype.walk = function () {
+  if (!this.isAlive) {
+    throw new Error("(*_*) Your pet has died.");
+  }
+  this.fitness = Math.min(this.fitness + 4, maxFitness);
+};
+
+Pet.prototype.feed = function () {
+  if (!this.isAlive) {
+    throw new Error("(*_*) Your pet has died.");
+  }
+  this.hunger = Math.max(this.hunger - 3, minHunger);
+};
+
+Pet.prototype.checkUp = function () {
+  if (!this.isAlive) {
+    throw new Error("(*_*) Your pet has died.");
+  }
+  if (
+    this.fitness <= minAcceptableFitness &&
+    this.hunger >= minAcceptableHunger
+  ) {
+    return "I am hungry AND I need a walk!";
+  }
+  if (this.fitness <= minAcceptableFitness) {
+    return "I need a walk!";
+  }
+  if (this.hunger >= minAcceptableHunger) {
+    return "I am hungry!";
+  }
```

Comment:
you've done the checks multiple times in this function. You can save these results to a variable to make your code DRY-er, e.g.
```
...
const needsWalk = this.fitness <= minAcceptableFitness;
const isHungry = this.hunger >= minAcceptableHunger;

if (needsWalk && isHungry) return "I am hungry AND I need a walk!";
if (needsWalk) return "I need a walk!";
if (isHungry) return "I am hungry!";
...
```

Reply:
Thanks for the feedback Perry. I've included this lines of code in pet.js but now that I run npm test I'm getting failed messages: 
<img width="958" alt="image" src="https://github.com/PerryBaran/virtual-pet-armghanz/assets/124094971/09017670-7eb0-4d10-b3c0-34a7554f26b4">